docs/application/native/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1134 @@
+  {
+      int ret;
+      ret = vc_mgr_set_recognition_mode(mode);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get or set private data between voice control manager and voice control engine using `vc_mgr_get_private_data()` and `vc_mgr_set_private_data()`.
+  `vc_mgr_get_private_data()` is used when the parameters move from voice control engine to voice control manager. `vc_mgr_set_private_data()` is used when the parameters move from voice control manager to voice control engine. These functions are used when the voice control manager is in the `VC_STATE_READY` state:
+
+  ```c
+  /* Get */
+  void get_private_data(const char* private_key)
+  {
+      int ret;
+      char* data = NULL;
+      ret = vc_mgr_get_private_data(private_key, &data);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Set */
+  void set_private_data(const char* private_key, const char* private_data)
+  {
+      int ret;
+      ret = vc_mgr_set_private_data(private_key, private_data);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+<a name="commands"></a>
+## Manage commands
+
+You can use command list to manage the commands. You can add or remove the commands to the command list and retrieve the command information using the command list.
+
+To create a command list and commands:
+
+1.  Create a command list with a command list handle.
+    The command list can have multiple commands. Each command has a text and a type.
+    The group can have `VC_COMMAND_TYPE_BACKGROUND`, `VC_COMMAND_TYPE_EXCLUSIVE`, `VC_COMMAND_TYPE_FOREGROUND`, `VC_COMMAND_TYPE_SYSTEM`, `VC_COMMAND_TYPE_SYSTEM_BACKGROUND`, and `VC_COMMAND_TYPE_WIDGET` type commands:
+
+    > [!NOTE]
+    > The order of command group priority is `VC_COMMAND_TYPE_SYSTEM`, `VC_COMMAND_TYPE_WIDGET`, `VC_COMMAND_TYPE_FOREGROUND`, `VC_COMMAND_TYPE_SYSTEM_BACKGROUND`, and `VC_COMMAND_TYPE_BACKGROUND`. The `VC_COMMAND_TYPE_EXCLUSIVE` type is used in special situations.
+    > Normally, the `VC_COMMAND_TYPE_SYSTEM` and `VC_COMMAND_TYPE_SYSTEM_BACKGROUND` commands are only registered in the voice control manager.
+    > The `VC_COMMAND_TYPE_FOREGROUND` and `VC_COMMAND_TYPE_BACKGROUND` commands are used in the voice control client application.
+    > The `VC_COMMAND_TYPE_WIDGET` command is automatically registered in the elementary on the screen.
+    > The `VC_COMMAND_TYPE_EXCLUSIVE` commands have special priority, so these are used when recording starts with `vc_mgr_start(true)`.
+    > When recording starts with `vc_mgr_start(true)`, commands having other priorities are not recognized, only `VC_COMMAND_TYPE_EXCLUSIVE` commands are recognized.
+
+    ```c
+    void
+    create_command_list()
+    {
+        int ret;
+        vc_cmd_list_h vc_cmd_list;
+        ret = vc_cmd_list_create(&vc_cmd_list);
+        if (VC_ERROR_NONE != ret)
+            /* Error handling */
+    }
+    ```
+
+2.  Create a command.
```

Comment:
```suggestion
2.  Create a command, command handle, and then define the command and its type:

Reply:
To create and define a voice command, user creates the command handle and set the command text and type.

So,, how about this?
```suggestion
2.  Create a command with a command handle, and then set the command text and its type:
```